app/services/import_service.rb

```diff
@@ -30,6 +32,33 @@ def import_follows!
     import_relationships!('follow', 'unfollow', @account.following, ROWS_PROCESSING_LIMIT, reblogs: { header: 'Show boosts', default: true }, notify: { header: 'Notify on new posts', default: false }, languages: { header: 'Languages', default: nil })
   end
 
+  def import_tag_follows!
+    parse_import_data!(['Tag'])
+    names = @data.take(ROWS_PROCESSING_LIMIT).map { |row| Tag.normalize(row['Tag'].strip) }.drop(1)
+
+    found_tags = Tag.matching_name(names)
+    found_names = found_tags.map(&:name)
+    missing_names = names - found_names
+
+    follow_tags = []
+
+    ApplicationRecord.transaction do
```

Comment:
I hope I'm not over commenting here.
What would you think about doing something like that?

```suggestion
    missing_names = names - found_tags.map(&:name)

    Tag.transaction do
      # I think Tag#find_or_create_by_names returns an array, and can (optionally) yield to a ablock, but mustn't
      follow_tags = Tag.find_or_create_by_names(missing_names)

      TagFollow.transaction do
        if @import.overwrite?
          @account.tag_relationships.destroy_all
          follow_tags.concat(found_tags)
        end
  
        follow_tags.each do |tag|
          @account.tag_relationships.create!(tag: tag)
        end
      end
    end
```